### PR TITLE
[FEATURE] - Generic file writer

### PR DESCRIPTION
**Pull request by riccamini**
Generic file writer

## Description
This PR provides a generic FileWriter class for several file formats and can be configured with any required option

## Related Issue
https://github.com/Nike-Inc/koheesio/issues/52

## Motivation and Context
Koheesio is currently shipping with the FileLoader which allows to read any file format specifying any needed option. The same feature, on the other hand, is missing on the writer side as there's no current class that supports generic file writing

## How Has This Been Tested?
Unit tests with mocks for DataFrameWriter calls

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/koheesio/spark/writers/file_writer.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    text = "text"
+
+
+class FileWriter(Writer, ExtraParamsMixin):
+    """
+    A configurable Writer that allows writing to different file formats providing any option needed.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:41]: For formatting purposes, change this to a header:
~~~markdown
Examples
--------
...
~~~

File: src/koheesio/spark/writers/file_writer.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        ```python
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:42]: ... and also decent this one level once you change to header as above

File: src/koheesio/spark/writers/file_writer.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class CsvFileWriter(FileWriter):
+    """Writes a DataFrame to a CSV file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.csv`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:78]: Same here

File: src/koheesio/spark/writers/file_writer.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class ParquetFileWriter(FileWriter):
+    """Writes a DataFrame to a Parquet file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.parquet`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:94]: Etc.

File: src/koheesio/spark/writers/file_writer.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class AvroFileWriter(FileWriter):
+    """Writes a DataFrame to an Avro file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.avro`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:111]: Etc.

File: src/koheesio/spark/writers/file_writer.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class JsonFileWriter(FileWriter):
+    """Writes a DataFrame to a JSON file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.json`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:127]: Etc.

File: src/koheesio/spark/writers/file_writer.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class OrcFileWriter(FileWriter):
+    """Writes a DataFrame to an ORC file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.orc`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:143]: Etc.

File: src/koheesio/spark/writers/file_writer.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class TextFileWriter(FileWriter):
+    """Writes a DataFrame to a text file.
+
+    This class is a convenience class that sets the `format` field to `FileFormat.text`.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/file_writer.py:159]: Etc.

**Review by dannymeijer — CHANGES_REQUESTED** [review]

**Comment by riccamini** [comment]
@dannymeijer Hope now it renders well

**Review by dannymeijer — APPROVED** [review]

**Review by pariksheet — APPROVED** [review]
Cool stuff.. LGTM

**Pull request merged**